app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/effects/DeleteFiles.kt

```diff
@@ -40,18 +40,31 @@ data class DeleteFiles(private val booksOnDiskListItem: List<BookOnDisk>) :
 
   override fun invokeWith(activity: AppCompatActivity) {
     (activity as ZimManageActivity).cachedComponent.inject(this)
+
+    var name = "\n"
     booksOnDiskListItem.forEach {
-      dialogShower.show(DeleteZim(it), {
+      name += "\n" + it.book.title
+    }
+
+    dialogShower.show(DeleteZims(name), {
+      var allDeleted = true
+
+      booksOnDiskListItem.forEach {
         if (deleteSpecificZimFile(it)) {
           if (it.file.canonicalPath == zimReaderContainer.zimCanonicalPath) {
             zimReaderContainer.setZimFile(null)
           }
-          activity.toast(R.string.delete_specific_zim_toast)
         } else {
-          activity.toast(R.string.delete_zim_failed)
+          allDeleted = false
         }
-      })
-    }
+      }
+
+      if (allDeleted) {
```

Comment:
If 2 execution branches are the same aside from the parameters then we can move the if branch into the method call and avoid repeating `activity.toast` twice

Reply:
@macgills I'm sorry, I did not get this. Which method call should I move the if branch into?
I was thinking of using a ternary operator for this purpose instead.

Reply:
yes that was my point but without any variable.
```
activity.toast( if(condition) R.something else R.somethingElse)
```

Reply:
Done.